decidim-participatory_processes/app/models/decidim/participatory_process.rb

```diff
@@ -57,6 +58,12 @@ class ParticipatoryProcess < ApplicationRecord
     scope :upcoming, -> { where(arel_table[:start_date].gt(Time.current)) }
     scope :active, -> { where(arel_table[:start_date].lteq(Time.current).and(arel_table[:end_date].gt(Time.current).or(arel_table[:end_date].eq(nil)))) }
 
+    scope :visible_for, lambda { |user|
+                          joins("LEFT JOIN decidim_participatory_space_private_users ON
+                          decidim_participatory_space_private_users.privatable_to_id = decidim_participatory_processes.id")
```

Comment:
I think this scope can be moved to the `Decidim::HasPrivateUsers` if you change `decidim_participatory_processes.id` to `arel_table[:id]` (haven't checked though), but this logic is duplicated in `Assembly`. If we could extract it to the concern it we super good 👌

Reply:
changed `decidim_participatory_processes ` to `#{table_name}`